2020-08-12 09:42:34,747 : INFO : Start of Training
2020-08-12 09:42:34,747 : INFO : Run_id:2020-08-12_094234_538201314
2020-08-12 09:42:34,747 : INFO : Start of Data Load, validation and transformation
2020-08-12 09:42:34,747 : INFO : Start of Archiving Old Rejected Files...
2020-08-12 09:42:34,748 : INFO : End of Archiving Old Rejected Files...
2020-08-12 09:42:34,748 : INFO : Start of Archiving Old Validated Files...
2020-08-12 09:42:34,751 : INFO : End of Archiving Old Validated Files...
2020-08-12 09:42:34,751 : INFO : Start of Archiving Old Processed Files...
2020-08-12 09:42:34,753 : INFO : End of Archiving Old Processed Files...
2020-08-12 09:42:34,753 : INFO : Start of Archiving Old Result Files...
2020-08-12 09:42:34,754 : INFO : End of Archiving Old Result Files...
2020-08-12 09:42:34,754 : INFO : Start of Reading values From Schema...
2020-08-12 09:42:34,755 : INFO : End of Reading values From Schema...
2020-08-12 09:42:34,755 : INFO : Start of Validating Column Length...
2020-08-12 09:42:34,755 : INFO : End of Validating Column Length...
2020-08-12 09:42:34,755 : INFO : Start of Validating Missing Values...
2020-08-12 09:42:34,755 : INFO : End of Validating Missing Values...
2020-08-12 09:42:34,755 : INFO : Start of Replacing Missing Values with NULL...
2020-08-12 09:42:34,755 : INFO : End of Replacing Missing Values with NULL...
2020-08-12 09:42:34,755 : INFO : Start of Creating Table...
2020-08-12 09:42:34,756 : INFO : Opened training database successfully
2020-08-12 09:42:34,756 : INFO : Tables created successfully
2020-08-12 09:42:34,757 : INFO : Closed training database successfully
2020-08-12 09:42:34,757 : INFO : End of Creating Table...
2020-08-12 09:42:34,757 : INFO : Opened training database successfully
2020-08-12 09:42:34,757 : INFO : Start of Inserting Data into Table...
2020-08-12 09:42:34,757 : INFO : End of Inserting Data into Table...
2020-08-12 09:42:34,757 : INFO : Start of Exporting Data into CSV...
2020-08-12 09:42:34,759 : INFO : Opened training database successfully
2020-08-12 09:42:34,768 : INFO : End of Exporting Data into CSV...
2020-08-12 09:42:34,769 : INFO : Start of Moving Processed Files...
2020-08-12 09:42:34,769 : INFO : End of Moving Processed Files...
2020-08-12 09:42:34,769 : INFO : End of Data Load, validation and transformation
2020-08-12 09:42:34,769 : INFO : Start of Preprocessing...
2020-08-12 09:42:34,769 : INFO : Start of reading dataset...
2020-08-12 09:42:34,778 : INFO : End of reading dataset...
2020-08-12 09:42:34,779 : INFO : Start of Droping Columns...
2020-08-12 09:42:34,781 : INFO : End of Droping Columns...
2020-08-12 09:42:34,781 : INFO : Start of replacing invalid values...
2020-08-12 09:42:34,857 : INFO : end of replacing invalid values...
2020-08-12 09:42:34,857 : INFO : Start of encode Categorical Values ...
2020-08-12 09:42:34,868 : INFO : end of encode Categorical Values...
2020-08-12 09:42:34,868 : INFO : Start of feature encoding...
2020-08-12 09:42:34,889 : INFO : End of feature encoding...
2020-08-12 09:42:34,893 : INFO : Start of Droping Columns...
2020-08-12 09:42:34,896 : INFO : End of Droping Columns...
2020-08-12 09:42:34,897 : INFO : Start of finding missing values...
2020-08-12 09:42:34,917 : INFO : End of finding missing values...
2020-08-12 09:42:34,917 : INFO : Start of imputing missing values...
2020-08-12 09:42:34,925 : INFO : End of imputing missing values...
2020-08-12 09:42:34,926 : INFO : Start of splitting features and label ...
2020-08-12 09:42:34,927 : INFO : End of splitting features and label ...
2020-08-12 09:42:34,928 : INFO : End of Preprocessing...
2020-08-12 09:42:34,929 : INFO : Start of elbow plotting...
2020-08-12 09:42:35,474 : INFO : The optimum number of clusters is: 3
2020-08-12 09:42:35,474 : INFO : End of elbow plotting...
2020-08-12 09:42:35,475 : INFO : Start of Create clusters...
2020-08-12 09:42:35,506 : INFO : Start of Save Models
2020-08-12 09:42:35,506 : INFO : Start of Save Models
2020-08-12 09:42:35,510 : INFO : Model File KMeans saved
2020-08-12 09:42:35,510 : INFO : Model File KMeans saved
2020-08-12 09:42:35,510 : INFO : End of Save Models
2020-08-12 09:42:35,510 : INFO : End of Save Models
2020-08-12 09:42:35,511 : INFO : succesfully created 3clusters.
2020-08-12 09:42:35,511 : INFO : End of Create clusters...
2020-08-12 09:42:35,516 : INFO : Start of finding best model...
2020-08-12 09:42:35,516 : INFO : Start of finding best params for KNN algo...
2020-08-12 09:42:44,495 : INFO : Knn Forest best params: {'algorithm': 'ball_tree', 'leaf_size': 10, 'n_neighbors': 2, 'p': 1}
2020-08-12 09:42:44,495 : INFO : End of finding best params for knn algo...
2020-08-12 09:42:44,646 : INFO : AUC for knn:0.5877568493150684
2020-08-12 09:42:44,647 : INFO : Start of finding best params for randomforest algo...
2020-08-12 09:43:06,411 : INFO : Random Forest best params: {'criterion': 'gini', 'max_depth': 3, 'max_features': 'log2', 'n_estimators': 50}
2020-08-12 09:43:06,411 : INFO : End of finding best params for randomforest algo...
2020-08-12 09:43:06,418 : INFO : AUC for Random Forest:0.858304794520548
2020-08-12 09:43:06,418 : INFO : End of finding best model...
2020-08-12 09:43:06,418 : INFO : Start of Save Models
2020-08-12 09:43:06,418 : INFO : Start of Save Models
2020-08-12 09:43:06,421 : INFO : Model File RandomForest0 saved
2020-08-12 09:43:06,421 : INFO : Model File RandomForest0 saved
2020-08-12 09:43:06,422 : INFO : End of Save Models
2020-08-12 09:43:06,422 : INFO : End of Save Models
2020-08-12 09:43:06,426 : INFO : Start of finding best model...
2020-08-12 09:43:06,426 : INFO : Start of finding best params for KNN algo...
2020-08-12 09:43:41,535 : INFO : Knn Forest best params: {'algorithm': 'ball_tree', 'leaf_size': 10, 'n_neighbors': 1, 'p': 1}
2020-08-12 09:43:41,536 : INFO : End of finding best params for knn algo...
2020-08-12 09:43:41,657 : INFO : Accuracy for knn:1.0
2020-08-12 09:43:41,658 : INFO : Start of finding best params for randomforest algo...
2020-08-12 09:43:41,665 : ERROR : Exception raised while finding best params for randomforest algo:Cannot have number of splits n_splits=5 greater than the number of samples: n_samples=3.
Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 56, in best_params_randomforest
    # Creating an object of the Grid Search class
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 710, in fit
    self._run_search(evaluate_candidates)
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 1151, in _run_search
    evaluate_candidates(ParameterGrid(self.param_grid))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 689, in evaluate_candidates
    cv.split(X, y, groups)))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_split.py", line 333, in split
    .format(self.n_splits, n_samples))
ValueError: Cannot have number of splits n_splits=5 greater than the number of samples: n_samples=3.
2020-08-12 09:43:41,669 : ERROR : Exception raised while finding best model:
Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 56, in best_params_randomforest
    # Creating an object of the Grid Search class
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 710, in fit
    self._run_search(evaluate_candidates)
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 1151, in _run_search
    evaluate_candidates(ParameterGrid(self.param_grid))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 689, in evaluate_candidates
    cv.split(X, y, groups)))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_split.py", line 333, in split
    .format(self.n_splits, n_samples))
ValueError: Cannot have number of splits n_splits=5 greater than the number of samples: n_samples=3.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 194, in get_best_model
    self.logger.info('AUC for knn:' + str(self.knn_score))
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 75, in best_params_randomforest
    return self.rfc
Exception
2020-08-12 09:43:41,671 : ERROR : Unsuccessful End of Training
Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 56, in best_params_randomforest
    # Creating an object of the Grid Search class
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 710, in fit
    self._run_search(evaluate_candidates)
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 1151, in _run_search
    evaluate_candidates(ParameterGrid(self.param_grid))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_search.py", line 689, in evaluate_candidates
    cv.split(X, y, groups)))
  File "C:\Users\sreeh\AppData\Roaming\Python\Python36\site-packages\sklearn\model_selection\_split.py", line 333, in split
    .format(self.n_splits, n_samples))
ValueError: Cannot have number of splits n_splits=5 greater than the number of samples: n_samples=3.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 194, in get_best_model
    self.logger.info('AUC for knn:' + str(self.knn_score))
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 75, in best_params_randomforest
    return self.rfc
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\training\train_model.py", line 78, in training_model
    best_model_name, best_model = self.modelTuner.get_best_model(x_train, y_train, x_test, y_test)
  File "C:\Users\sreeh\Desktop\group project\actual project\apps\tuning\model_tuner.py", line 215, in get_best_model
Exception
